tasks/main.yml

```diff
@@ -78,61 +87,61 @@
 # be mentioned explicitly in the output of the command "systemctl
 # status mongod.service".
 - name: Test for AssertPathIsMountPoint lines
-  command: grep -c ^AssertPathIsMountPoint= /lib/systemd/system/mongod.service
-  register: test_assertpathismountpoint
-  ignore_errors: yes
+  ansible.builtin.command: grep -c ^AssertPathIsMountPoint= /lib/systemd/system/mongod.service
   changed_when: false
+  ignore_errors: yes
+  register: test_assertpathismountpoint
 - name: Make sure mongo data file system is mounted before starting
-  lineinfile:
+  ansible.builtin.lineinfile:
     dest: /lib/systemd/system/mongod.service
     insertafter: ^RequiresMountsFor=
-    state: present
     line: AssertPathIsMountPoint=/var/lib/mongodb
+    state: present
   when: test_assertpathismountpoint.rc != 0
 - name: Make sure mongo journal file system is mounted before starting
-  lineinfile:
+  ansible.builtin.lineinfile:
     dest: /lib/systemd/system/mongod.service
     insertafter: ^RequiresMountsFor=
-    state: present
     line: AssertPathIsMountPoint=/var/lib/mongodb/journal
+    state: present
   when: test_assertpathismountpoint.rc != 0
 - name: Make sure mongo log file system is mounted before starting
-  lineinfile:
+  ansible.builtin.lineinfile:
     dest: /lib/systemd/system/mongod.service
     insertafter: ^RequiresMountsFor=
-    state: present
     line: AssertPathIsMountPoint=/var/log/mongodb
+    state: present
   when: test_assertpathismountpoint.rc != 0
 
 # /run/mongodb (AKA /var/run/mongodb) is used to store a file
 # containing the mongod pid.  We shouldn't start up until that
 # directory exists and has the correct permissions.
 - name: Test for RuntimeDirectory line
-  command: grep -c ^RuntimeDirectory= /lib/systemd/system/mongod.service
-  register: test_runtimedirectory
-  ignore_errors: yes
+  ansible.builtin.command: grep -c ^RuntimeDirectory= /lib/systemd/system/mongod.service
   changed_when: false
+  ignore_errors: yes
+  register: test_runtimedirectory
 - name: Make sure /run/mongodb is created before starting
```

Comment:
Same question about using a `block` for these `when: test_runtimedirectory.rc != 0` tasks.

Reply:
Those `when` clauses for those tasks are different, are they not?

Reply:
Yep, missed the `mode` at the end of the second test.